ansible_collections/arista/avd/roles/eos_designs/python_modules/l3_edge/router_bgp.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdMissingVariableError
+
+from .utils import UtilsMixin
+
+
+class RouterBgpMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def router_bgp(self) -> dict | None:
+        """
+        Return structured config for router_bgp
+        """
+
+        if not self._underlay_bgp:
+            return None
+
+        neighbors = {}
+        neighbor_interfaces = {}
+        for p2p_link in self._filtered_p2p_links:
+            if not (p2p_link.get("include_in_underlay_protocol") is True):
+                continue
+
+            if p2p_link["data"]["bgp_as"] is None or p2p_link["data"]["peer_bgp_as"] is None:
+                raise AristaAvdMissingVariableError("l3_edge.p2p_links.[].as")
+
+            neighbor = {
+                "remote_as": p2p_link["data"]["peer_bgp_as"],
+                "description": p2p_link["data"]["peer"],
+                "peer_group": self._peer_group_ipv4_underlay_peers_name,
+            }
+
+            if self._underlay_rfc5549:
+                # RFC5549
```

Comment:
would make sense to have this comment indented the same as `# Regular BGP Neighbors` below

Reply:
done